### PR TITLE
feat: Allow configuring export macros to not reply

### PR DESCRIPTION
**Pull request by adamspofford-dfinity**
Adds a `reply` field to `ic_cdk_macros`'s export macros, which controls whether `ic0.msg_reply` is called by the macro. If it is not, the canister function must (and therefore can) call that function itself. 

Also adds an `Empty` type to `ic_cdk::api::call`, for annotating not-physically-returning functions with the type they semantically return, for use with the former feature.

### CODE REVIEW

**Review by ericswanson-dfinity — COMMENTED** [review]
This approach looks great.  Can we add an example and a github workflow that uses the example to test the functionality?

**Comment by adamspofford-dfinity** [comment]
Sure, I'll just change the ones I was going to change in the next PR instead.

File: docs/modules/rust-guide/examples/profile-tutorial/profile.rs [hunk 1]
```diff
@@ -45,22 +51,26 @@ fn update(profile: Profile) {
...
-            return Some(p);
+            profile = Some(p);
+            break;
         }
 
         for x in p.keywords.iter() {
             if x.to_lowercase() == text {
-                return Some(p);
+                profile = Some(p);
+                break;
```
Comment (ericswanson-dfinity) [inline docs/modules/rust-guide/examples/profile-tutorial/profile.rs:69]: Does this need to break out of the outer `for` loop?

Comment (adamspofford-dfinity) [reply docs/modules/rust-guide/examples/profile-tutorial/profile.rs:69]: Whoops!

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**